addons/cloud_storage_google/models/cloud_storage_google.py

```diff
@@ -0,0 +1,92 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import re
+from urllib.parse import unquote, quote
+
+try:
+    from google.oauth2 import service_account
+    from google.auth.transport.requests import Request
+except ImportError:
+    service_account = Request = None
+
+from odoo import models
+from odoo.exceptions import ValidationError
+
+from ..utils.cloud_storage_google_utils import generate_signed_url_v4
+
+GOOGLE_CLOUD_STORAGE_ENDPOINT = 'https://storage.googleapis.com'
+
+
+class CloudStorageGoogle:
+    _url_pattern = re.compile(rf'{GOOGLE_CLOUD_STORAGE_ENDPOINT}/(?P<bucket_name>[\w\-.]+)/(?P<blob_name>[^?]+)')
```

Comment:
Don't you have to escape the dots `"."` coming from `GOOGLE_CLOUD_STORAGE_ENDPOINT` in this regex?